.dev_scripts/benchmark/benchmark_regression.py

```diff
@@ -0,0 +1,250 @@
+                cur_cpus_per_task = model[mode][
+                    'cpus_per_task'] if 'cpus_per_task' in model[mode].keys(
+                    ) else args.cpus_per_task
+
+                cur_task_name = mode + '_' + task_name
+                cur_work_dir = osp.join(args.root_work_dir, cur_task_name)
+
+                if mode == 'test':
+                    # deal with extra python arguments
+                    py_cmd = f' --work-dir {cur_work_dir} '
+                    if 'py_args' in model[mode].keys():
+                        keys = list(model[mode]['py_args'].keys())
+                        values = list(model[mode]['py_args'].values())
+
+                        for k in range(len(keys)):
+                            if values[k] is None:
+                                if keys[k] in ['fuse_conv_bn', 'gpu_collect']:
+                                    py_cmd += f' --{keys[k]} '
+                            else:
+                                py_cmd += f' --{keys[k]} {values[k]} '
+                    cmd = f'MASTER_PORT={cur_port} GPUS={cur_gpus} ' + \
+                          f'GPUS_PER_NODE={cur_gpus_per_node} ' + \
+                          f'CPUS_PER_TASK={cur_cpus_per_task} ' + \
+                          f'./tools/slurm_test.sh {cur_partition} ' + \
+                          f'{cur_task_name} ' + \
+                          f'{cur_config} {cur_checkpoint} ' + \
+                          f'{py_cmd}'
+
+                    os.system(f'tmux send-keys "{cmd}" "C-m"')
+
+                else:
+                    py_cmd = ' '
+                    # deal with extra python arguments
+                    if 'py_args' in model[mode].keys():
+                        keys = list(model[mode]['py_args'].keys())
+                        values = list(model[mode]['py_args'].values())
+
+                        for k in range(len(keys)):
+                            if values[k] is None:
+                                if keys[k] in [
+                                        'no-validate', 'deterministic',
+                                        'autoscale-lr'
+                                ]:
+                                    py_cmd += f' --{keys[k]} '
+                            else:
+                                py_cmd += f' --{keys[k]} {values[k]} '
+                    cmd = f'MASTER_PORT={cur_port} GPUS={cur_gpus} ' + \
+                          f'GPUS_PER_NODE={cur_gpus_per_node} ' + \
+                          f'CPUS_PER_TASK={cur_cpus_per_task} ' + \
+                          f'./tools/slurm_train.sh {cur_partition} ' + \
+                          f'{cur_task_name} ' + \
+                          f'{cur_config} {cur_work_dir} ' + \
+                          f'{py_cmd}'
+                    os.system(f'tmux send-keys "{cmd}" "C-m"')
+
+                cur_port += 1
+                # if the port is used, use a random number for port
+                while not is_port_available(cur_port, ip):
```

Comment:
How about wrap L236-237 into a util function `get_available_port(range=(29000, 39000))`?
Maybe we can also improve the train.py and test.py?